pipeline/scripts/15-dIEM_violin.R

```diff
@@ -0,0 +1,470 @@
+# For untargeted metabolomics, this tool calculates probability scores for 
+# metabolic disorders. In addition, it provides visual support with violin plots 
+# of the DIMS measurements for the lab specialists.
+# Input needed: 
+# 1. Excel file in which metabolites are listed with their intensities for
+#    controls (with C in samplename) and patients (with P in samplename) and their
+#    corresponding Z-scores. 
+# 2. All files from github: https://github.com/UMCUGenetics/dIEM
+
+library(dplyr) # tidytable is for other_isobaric.R (left_join)
+library(reshape2) # used in prepare_data.R
+library(openxlsx) # for opening Excel file
+library(ggplot2) # for plotting
+library(gridExtra) # for table top highest/lowest
+
+# load functions
+#source("/hpc/dbg_mz/production/DIMS/pipeline/scripts/AddOnFunctions/same_samplename.R")
+#source("/hpc/dbg_mz/production/DIMS/pipeline/scripts/AddOnFunctions/prepare_data.R")
+#source("/hpc/dbg_mz/production/DIMS/pipeline/scripts/AddOnFunctions/prepare_data_perpage.R")
+#source("/hpc/dbg_mz/production/DIMS/pipeline/scripts/AddOnFunctions/prepare_toplist.R")
+#source("/hpc/dbg_mz/production/DIMS/pipeline/scripts/AddOnFunctions/violin_plots.R")
+#source("/hpc/dbg_mz/production/DIMS/pipeline/scripts/AddOnFunctions/prepare_alarmvalues.R")
+#temporary:
+source("/hpc/dbg_mz/development/DIMS_Violinplots/pipeline/scripts/AddOnFunctions/same_samplename.R")
+source("/hpc/dbg_mz/development/DIMS_Violinplots/pipeline/scripts/AddOnFunctions/prepare_data.R")
+source("/hpc/dbg_mz/development/DIMS_Violinplots/pipeline/scripts/AddOnFunctions/prepare_data_perpage.R")
+source("/hpc/dbg_mz/development/DIMS_Violinplots/pipeline/scripts/AddOnFunctions/prepare_toplist.R")
+source("/hpc/dbg_mz/development/DIMS_Violinplots/pipeline/scripts/AddOnFunctions/violin_plots.R")
+source("/hpc/dbg_mz/development/DIMS_Violinplots/pipeline/scripts/AddOnFunctions/prepare_alarmvalues.R")
```

Comment:
For these temporary / development paths, it might be better to use a config file or something similar.

Reply:
The problem is that the pipeline is started from the GUI, without the need for a user to log into the HPC, so I don't know how to add a config file in this case. Will be solved in the Nextflow edition.